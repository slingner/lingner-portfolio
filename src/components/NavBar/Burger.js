import React from 'react';
import styled from 'styled-components';
import { colors } from '../constants';

const StyledBurger = styled.button`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  /* position: absolute; */
  width: 2rem;
  height: 2.2em;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  z-index: 10;
  margin-left: 10px;

  &:focus {
    outline: none;
  }

  div {
    /* padding: 10px; */
    width: 1.5rem;
    height: 0.5em;
    background-color: ${colors.darkgrey};
    border-radius: 1px;
    transition: all 0.3s linear;
    /* position: fixed; */
    transform-origin: 1px;

    :first-child {
      transform: ${({ open }) => (open ? 'rotate(45deg)' : 'rotate(0)')};
      background: ${({ open }) => (open ? colors.white : colors.darkgrey)};
    }

    :nth-child(2) {
      opacity: ${({ open }) => (open ? '0' : '1')};
      transform: ${({ open }) => (open ? 'translateX(20px)' : 'translateX(0)')};
    }

    :nth-child(3) {
      transform: ${({ open }) => (open ? 'rotate(-45deg)' : 'rotate(0)')};
      background: ${({ open }) => (open ? colors.white : colors.darkgrey)};
    }
  }
`;

const Burger = (props) => {
  return (
    <StyledBurger open={props.open}>
      <div />
      <div />
      <div />
    </StyledBurger>
  );
};

export default Burger;
