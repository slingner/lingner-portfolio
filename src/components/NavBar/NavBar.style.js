import styled from 'styled-components/macro';
import { colors } from '../constants';
import { Link } from 'react-router-dom';

export const NavWrapper = styled.div`
  justify-content: center;
  height: ${({ open }) => (open ? '100%' : '0')};
  width: ${({ open }) => (open ? '100%' : '0')};
  position: fixed;
  top: 0;
  left: 0;
  z-index: 18;
  background: ${colors.darkgrey};

  .mobile {
    background-color: ${({ open }) => (open ? colors.darkgrey : 'transparent')};
  }

  @media screen and (max-width: 768px) {
    flex-direction: column;
  }
`;

export const MenuLink = styled(Link)`
  font-size: 16px;
  color: ${colors.white};
  text-decoration: none;
  margin-left: 25px;
  align-items: flex-end;

  :hover {
    cursor: pointer;
    color: ${colors.coral};
  }
`;

export const MobileButton = styled.div`
  border: none;
  background: transparent;
  padding: 15px;
  margin: 5px;

  @media screen and (max-width: 768px) {
    display: inline-block;
    height: 40px;
  }

  :hover {
    cursor: pointer;
  }
`;
