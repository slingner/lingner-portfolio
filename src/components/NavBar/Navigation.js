/* eslint-disable react/prop-types */
import React from 'react';
import styled from 'styled-components/macro';
import { colors } from '../constants';
import { keyframes } from 'styled-components';
import { fadeOutUp } from 'react-animations';

const fadeOutUpAnimation = keyframes`${fadeOutUp}`;

const NavLinks = styled.ul`
  list-style: none;
  color: ${(props) => (props.color ? props.color : colors.white)};
  margin-right: 10%;
  margin-top: 5%;
  text-align: right;
  display: flex;
  flex-direction: ${(props) => (props.direction ? props.direction : 'column')};

  @media (min-width: 600px) {
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
  }
`;

export const NavItem = styled.a`
  color: ${(props) => (props.color ? props.color : colors.white)};
  text-decoration: none;
  margin: 0px auto;
  font-size: 60px;
  font-family: 'Raleway', sans-serif;
  font-display: swap;

  :hover {
    cursor: pointer;
    color: ${colors.blue};
    animation: 0.5s ${fadeOutUpAnimation};
  }

  @media (max-width: 600px) {
    text-align: left;
    font-size: 60px;
  }

  @media (max-width: 374px) {
    text-align: left;
    font-size: 60px;
  }
`;

const Navigation = (props) => {
  return (
    <NavLinks direction={props.direction}>
      <li className="nav-links" color={props.color}>
        <NavItem href="#about" onClick={props.toggleNav} color={props.color}>
          ABOUT
        </NavItem>
      </li>

      <li className="nav-links" color={props.color}>
        <NavItem href="#work" onClick={props.toggleNav} color={props.color}>
          PROJECTS
        </NavItem>
      </li>

      <li className="nav-links" color={props.color}>
        <NavItem href="#contact" onClick={props.toggleNav} color={props.color}>
          CONTACT
        </NavItem>
      </li>
      <li className="nav-links" color={props.color}>
        <NavItem
          href="https://scott-lingner-portfolio.s3-us-west-1.amazonaws.com/ScottLingner_Resume.pdf"
          target="_blank"
          rel="nofollow noopener noreferrer"
          onClick={props.toggleNav}
          color={props.color}
          // style={{
          //   fontSize: '40px',
          //   padding: '8px',
          //   borderBottom: '1px solid white',
          // }}
        >
          RESUME
        </NavItem>
      </li>
    </NavLinks>
  );
};

export default Navigation;
