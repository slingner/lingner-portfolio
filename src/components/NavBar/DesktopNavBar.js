import React from 'react';
import Navigation from './Navigation';
import styled from 'styled-components';
import { colors } from '../constants';

const DesktopNav = styled.nav`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  /* background: ${colors.darkgrey}; */
  /* color: ${colors.darkgrey}; */

  .nav-links{
    margin-left: 20px;

    
  @media screen and (max-width: 768px) {
      display: none;
    }
  }
  
  .ul {
    display:flex;
    flex-direction: row;
  }
`;

const DesktopNavBar = (props) => {
  return (
    <DesktopNav>
      <Navigation direction='row' marginLeft='25px'/>
    </DesktopNav>
  );
};

export default DesktopNavBar;