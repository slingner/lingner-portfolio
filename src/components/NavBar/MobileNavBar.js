/* eslint-disable react/prop-types */
import React from 'react';
import styled from 'styled-components/macro';
import { colors } from '../constants';
import Navigation from './Navigation';

const MobileNav = styled.nav`
  display: block;
  background-color: ${colors.darkgrey};
  height: 100%;
  width: 100%;
  text-align: center;
  color: white;
`;

const MobileNavBar = (props) => {
  return (
    <MobileNav displayMobileNavBar={props.displayMobileNavBar}>
      <p
        style={{
          padding: '20px',
          textAlign: 'left',
          borderBottom: '1px solid white',
          // borderWidth: '10px',
        }}
      >
        SCOTT LINGNER
      </p>

      <Navigation
        toggleNav={props.toggleNav}
        direction="column"
        color={colors.white}
      />
    </MobileNav>
  );
};

export default MobileNavBar;
