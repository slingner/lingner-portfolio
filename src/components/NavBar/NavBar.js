import React, { useState } from 'react';
import MobileNavBar from './MobileNavBar';
import Burger from './Burger';
import { NavWrapper, MobileButton } from './NavBar.style';

function NavBar() {
  const [mobileNav, setMobileNav] = useState(false);

  const toggleTrueFalse = () => setMobileNav(!mobileNav);

  return (
    <header>
      <NavWrapper open={mobileNav}>
        <div className="mobile" onClick={toggleTrueFalse}>
          <MobileButton type="button" onClick={toggleTrueFalse}>
            <Burger aria-label="Burger Nav Menu" open={mobileNav} />
          </MobileButton>
        </div>

        {mobileNav ? (
          <MobileNavBar
            toggleNav={toggleTrueFalse}
            displayMobileNavBar={mobileNav}
          />
        ) : null}
      </NavWrapper>
    </header>
  );
}

export default NavBar;
