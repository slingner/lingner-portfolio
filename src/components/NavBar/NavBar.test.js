import React from 'react';
import { NavWrapper } from './NavBar.style';
import renderer from 'react-test-renderer';

describe(`<NavBar />`, () => {
  it('renders UI as expected', () => {
    const tree = renderer.create(<NavWrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
