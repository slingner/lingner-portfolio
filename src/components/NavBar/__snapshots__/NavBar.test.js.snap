// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders UI as expected 1`] = `
<div
  className="NavBarstyle__NavWrapper-zh9t3p-0 Htmbc"
/>
`;
